lib/flipper/adapters/memcached.rb

```diff
@@ -0,0 +1,74 @@
+require 'dalli'
+
+module Flipper
+  module Adapters
+    class Memcached
+      include Flipper::Adapter
+      FeaturesKey =  :flipper_features
+
+      attr_accessor :cache
+      attr_accessor :name
+
+      def initialize(adapter, address = 'localhost:11211', options = {})
+        @adapter = adapter
+        @name = adapter.name
+        @cache = Dalli::Client.new(address, options)
+      end
+
+      def features
+        cached(FeaturesKey) do
+          @adapter.features
+        end
+      end
+
+      def add(feature)
+        @adapter.add(feature).tap do
+          @cache.delete(FeaturesKey)
+        end
+      end
+
+      def remove(feature)
+        @adapter.remove(feature).tap do
+          @cache.delete(FeaturesKey)
+        end
+      end
+
+      def clear(feature)
+        @adapter.clear(feature).tap do
+          @cache.delete(feature)
+        end
+      end
+
+      def flush
```

Comment:
I'd drop this method. Since we can pass in an instance, the instance can be used to flush in tests. New methods per adapter for adapters that are used in production can cause tricky issues (like using the method in middleware, then removing that adapter and then that method is gone and errors).